components/eamxx/src/share/io/tests/io_basic.cpp

```diff
@@ -27,7 +27,7 @@
 namespace scream {
 
 constexpr int num_output_steps = 5;
-constexpr Real FillValue = DEFAULT_FILL_VALUE;
+constexpr Real FillValue = constants::DefaultFillValue<float>().value;
```

Comment:
I think we could maybe move the testing of fill value in a separate test? I think it's clearer if we keep the `io_basic.cpp` tests...well...basic. :)

There may be some code duplication, but that's prob justified by the fact that it makes it easier to debug I/O failures, since we're not mixing up many use cases in one.